core-processor/src/main/java/io/micronaut/inject/beans/visitor/BeanIntrospectionWriter.java

```diff
@@ -495,13 +494,18 @@ private void writeIntrospectionClass(ClassWriterOutputVisitor classWriterOutputV
         final Type superType = Type.getType(AbstractInitializableBeanIntrospection.class);
 
         ClassWriter classWriter = new AptClassWriter(ClassWriter.COMPUTE_MAXS | ClassWriter.COMPUTE_FRAMES, visitorContext);
-        classWriter.visit(V17, ACC_SYNTHETIC | ACC_FINAL,
+        classWriter.visit(
+            V17,
+            ACC_SYNTHETIC | ACC_FINAL | ACC_PUBLIC,
```

Comment:
why does it need to be public now?

Reply:
is it because it's instantiated directly now?

Reply:
Yes